plugin/sampling/strategystore/adaptive/processor_test.go

```diff
@@ -393,8 +393,15 @@ func TestRunCalculationLoop_GetThroughputError(t *testing.T) {
 	p, err := newProcessor(cfg, "host", mockStorage, mockEP, metrics.NullFactory, logger)
 	require.NoError(t, err)
 	p.shutdown = make(chan struct{})
-	defer close(p.shutdown)
-	go p.runCalculationLoop()
+	done := make(chan struct{})
+	go func() {
+		p.runCalculationLoop()
+		close(done)
+	}()
+	defer func() {
+		close(p.shutdown)
+		<-done
```

Comment:
do we really need `done` sync here? as long as goroutines are not blocked they will exit and goleak will tolerate that, it does not expect immediate clean state when it's called.

Reply:
Ah thanks for flagging this. They are not required (test pass without it). I was also misunderstanding the goleak semantics.

I think the issue I was having is before https://github.com/jaegertracing/jaeger/pull/5310/commits/dfe7adc7fd34389e2e507d9512eb64b5e19048f7, these tests were failing because a goroutine could be blocked in time.Sleep.

I'll remove this.